runtime/lua/vim/shared.lua

```diff
@@ -201,10 +210,11 @@ end
 ---@param t table<T, any> (table) Table
 ---@return T[] (list) List of keys
 function vim.tbl_keys(t)
-  assert(type(t) == 'table', string.format('Expected table, got %s', type(t)))
+  vim.validate({ t = { t, 't' } })
+  --- @cast t table<any,any>
 
   local keys = {}
-  for k, _ in pairs(t) do
+  for k in pairs(t) do
```

Comment:
Is LuaLS complaining about the `_` ? Not sure it's an improvement to remove that explicitness everywhere.

Reply:
No it's not complaining, the `_` just isn't doing anything.

There are loads of places in the codebase where unused varargs are ignored without an explicit `_`. Singling out `pairs` doesn't make sense to me.